### PR TITLE
[Wallet] Merge back changes made for mx pilot

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

As part of my work on #1011, I needed the work I did for displaying the local currency value in the send flow which was only in the mx pilot branch. Since that work was intertwined with other changes there, it was simpler to merge everything back that made sense.

Notable changes:

- Show local currency values in more screens (only when LOCAL_CURRENCY_SYMBOL is set)
  - Enter amount as local currency
  - Send confirmation displays local currency
  - Payment request confirmation displays local currency
- Country issues (Not sure what they were, I need to check with @cmcewen)
- Hide country autocomplete on exact match
- Spanish copy changes
- Prevent dex crash 
- Make backup content scrollable 
- Fix backup key in wrong language 
- Turn on autofocus when scanning QR codes

### Tested

Ran the tests.

Changes with the local currency in the send flow:
image (6)
image (5)
image (4)
image (3)_censored

### Other changes

- Don't mock `country-data` anymore. There was a test failing because of this.
- Updated copy for the invite sms

### Related issues

- #1011

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/android/gradle.properties [hunk 1]
```diff
@@ -20,4 +20,4 @@
 CELO_RELEASE_STORE_FILE=celo-release-key.keystore
 CELO_RELEASE_KEY_ALIAS=celo-key-alias
 # Setting this manually based on version number until we have this deploying via Cloud Build
-VERSION_CODE=100400201
\ No newline at end of file
+VERSION_CODE=100400209
```
Comment (jeanregisser) [inline packages/mobile/android/gradle.properties:23]: Not exactly sure we want that change too. Please let me know.

Comment (jmrossy) [reply packages/mobile/android/gradle.properties:23]: This translate to version 1.4.2 deployment # 9
so presumably we had to deploy pilot a few times. 
Since I'll be upping it to 1.4.3 for monday, this change is okay

File: packages/mobile/src/backup/BackupComplete.tsx [hunk 2]
```diff
@@ -78,13 +81,18 @@ const styles = StyleSheet.create({
...
     backgroundColor: 'white',
     justifyContent: 'space-between',
-    flexDirection: 'column',
-    paddingHorizontal: 20,
   },
-  questionTextContainer: {
+  scrollContainer: {
+    // https://medium.com/@peterpme/taming-react-natives-scrollview-with-flex-144e6ff76c08
+    flexGrow: 1,
+    backgroundColor: 'white',
```
Comment (jmrossy) [inline packages/mobile/src/backup/BackupComplete.tsx]: Nit: should be `colors.background`

File: packages/mobile/src/backup/BackupPhrase.tsx [hunk 3]
```diff
@@ -85,7 +85,7 @@ const styles = StyleSheet.create({
   },
   scrollContainer: {
     paddingHorizontal: 20,
-    flex: 1,
+    flexGrow: 1,
```
Comment (jmrossy) [inline packages/mobile/src/backup/BackupPhrase.tsx:88]: Why this change?

File: packages/mobile/src/components/LineItemRow.tsx [hunk 4]
```diff
@@ -26,7 +26,7 @@ function LineItemRow({
         <Text style={style.lineItemText}>{title}</Text>
         {titleIcon}
       </View>
-      {!!amount && <Text style={style.lineItemText}>{currencySymbol + amount}</Text>}
+      {!!amount && <Text style={style.lineItemText}>{amount}</Text>}
```
Comment (jmrossy) [inline packages/mobile/src/components/LineItemRow.tsx:29]: I disagree with this change. It's confusing that the amounts are in local currency but the fees are in celo dollars. I think removing the currency symbol makes this even worse. WDYT?

File: packages/mobile/src/paymentRequest/PaymentRequestReviewCard.tsx [hunk 5]
```diff
@@ -36,42 +38,64 @@ const mapStateToProps = (state: RootState): StateProps => {
...
-      recipient,
-      address,
-      e164PhoneNumber,
-      t,
-      dollarBalance,
-      currency,
-      value,
-      comment,
-    } = this.props
+function PaymentRequestReviewCard({
```
Comment (jmrossy) [inline packages/mobile/src/paymentRequest/PaymentRequestReviewCard.tsx:41]: This is a non-trivial component. Should we memoize? Does redux's checks for if a component should re-render still work for FCs?

File: packages/mobile/src/send/SendAmount.tsx [hunk 6]
```diff
@@ -155,9 +160,25 @@ export class SendAmount extends React.Component<Props, State> {
...
+    const parsedInputAmount = parseInputAmount(this.state.amount)
+
+    let dollarsAmount
+    if (LOCAL_CURRENCY_SYMBOL) {
+      const { localCurrencyExchangeRate } = this.props
+      dollarsAmount =
+        convertLocalAmountToDollars(parsedInputAmount, localCurrencyExchangeRate) ||
+        new BigNumber('')
+    } else {
+      dollarsAmount = parsedInputAmount
```
Comment (jmrossy) [inline packages/mobile/src/send/SendAmount.tsx:176]: nit just return parsedInputAmount here instead of using temp var dollarsAmount

File: packages/mobile/src/send/SendAmount.tsx [hunk 7]
```diff
@@ -180,14 +201,14 @@ export class SendAmount extends React.Component<Props, State> {
...
-    const amount = parseInputAmount(this.state.amount)
+    const dollarsAmount = this.getDollarsAmount()
     const recipient = this.getRecipient()
     // TODO (Rossy) Remove address field from some recipient types.
     const recipientAddress = getAddressFromRecipient(recipient, this.props.e164NumberToAddress)
 
     const confirmationInput: ConfirmationInput = {
       recipient,
-      amount,
+      amount: new BigNumber(dollarsAmount.toPrecision(18, BigNumber.ROUND_UP)),
```
Comment (jmrossy) [inline packages/mobile/src/send/SendAmount.tsx]: what's going on here?

File: packages/react-components/components/MoneyAmount.tsx [hunk 8]
```diff
@@ -48,4 +50,12 @@ const style = StyleSheet.create({
     fontSize: 24,
     color: colors.darkSecondary,
   },
+  currencyCode: {
+    ...fontStyles.regular,
+    fontSize: 16,
+    lineHeight: 64,
+    color: '#BDBDBD',
```
Comment (jmrossy) [inline packages/react-components/components/MoneyAmount.tsx]: Use the values in colors.ts instead of hard coding please

File: packages/react-components/components/MoneyAmount.tsx [hunk 9]
```diff
@@ -48,4 +50,12 @@ const style = StyleSheet.create({
     fontSize: 24,
     color: colors.darkSecondary,
   },
+  currencyCode: {
+    ...fontStyles.regular,
+    fontSize: 16,
```
Comment (jmrossy) [inline packages/react-components/components/MoneyAmount.tsx]: The `body` font style already specifies this size and font family, consider using that instead.

File: packages/utils/src/countries.ts [hunk 10]
```diff
@@ -24,16 +24,44 @@ const EMPTY_COUNTRY: LocalizedCountry = {
...
+    .toLowerCase()
+    .trim()
+
+interface CountrySearch {
+  displayName: string
+  countryCode: string
+}
+
+// https://stackoverflow.com/questions/43118692/typescript-filter-out-nulls-from-an-array
+function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
```
Comment (jmrossy) [inline packages/utils/src/countries.ts]: Don't think this belongs here

**Review by jmrossy — COMMENTED** [review]
Thanks for handling the merge back to master! Sorry if some of these comments are not for your code changes

File: packages/mobile/src/backup/BackupPhrase.tsx (hunk 3 above)
Comment (jeanregisser) [reply packages/mobile/src/backup/BackupPhrase.tsx:88]: Looks like it's the same as explained in `BackupComplete` about https://medium.com/@peterpme/taming-react-natives-scrollview-with-flex-144e6ff76c08
I'll let @cmcewen give more context.

File: packages/mobile/src/paymentRequest/PaymentRequestReviewCard.tsx (hunk 5 above)
Comment (jeanregisser) [reply packages/mobile/src/paymentRequest/PaymentRequestReviewCard.tsx:41]: I think we're fine.
Changing it to a FC didn't affect the re-render. This component is still wrapped by `connect` and only re-renders when the mapped props actually change ([based on shallow equality comparisons](https://react-redux.js.org/api/connect#mapstatetoprops-state-ownprops-object)).
Internally it does contain a new hook which also listens to redux, but again it will only re-render if the value changes ([based on `===` reference comparison](https://react-redux.js.org/api/hooks#equality-comparisons-and-updates))

File: packages/react-components/components/MoneyAmount.tsx (hunk 8 above)
Comment (jeanregisser) [reply packages/react-components/components/MoneyAmount.tsx]: This is a new color used. I'll add it there.

File: packages/mobile/src/send/SendAmount.tsx (hunk 7 above)
Comment (jeanregisser) [reply packages/mobile/src/send/SendAmount.tsx]: Ah yes 😸  It definitely needs explaining.

This is needed to have the maximum supported precision when the initial amount was entered in the local currency. Otherwise the confirmation screen won't display the same local currency amount that was entered or the fee/send will fail because of a decimal part remaining.

Let's take an example.
- user enters `2` MXN (their local currency)
- user expects `2` MXN to be displayed in the confirmation screen
- technically the confirmation screens receives `2 / exchangeRate`, if we take `19.67` as the exchange rate, the dollar amount is then `0.10167768174885612608`. Given we want to display the local amount in the confirmation too, we need to convert again, `0.10167768174885612608 * 19.67 = 1.9999999999999999999936` which displays as `1.99 MXN` (which is not what the user expects) and also yields the following fee calculation error: `CalculateFee/Error calculating fee:invalid number value (arg="value", coderType="uint256", value="101677681748856126.08")` as `0.10167768174885612608` is converted to wei but still has a decimal part which is not accepted by the contract.
- that's why we limit the precision to 18 decimals and round up at this precision! This really only affects local currency. If the user enters `2.99` cUSD, it still displays `2.99` in the confirmation screen.

I thought about passing the original local amount to the confirmation screen but this complicates things and we would still have the problem once we display the transaction in the activity feed which doesn't have the original entered amount.

Let me know if you see an alternative. Also I think we should be able to derive the `18` magic number from a constant. Maybe from `WEI_PER_CELO`? Or is there another place where it's already defined?

File: packages/utils/src/countries.ts (hunk 10 above)
Comment (jeanregisser) [reply packages/utils/src/countries.ts]: Yes any recommendation?

File: packages/mobile/src/components/LineItemRow.tsx (hunk 4 above)
Comment (jeanregisser) [reply packages/mobile/src/components/LineItemRow.tsx:29]: We talked about this when I implemented this for the pilot and we couldn't agree, except that we could just try for the pilot.

Looks like we need more people involved to decide on this with the feedback from the pilot.

File: packages/utils/src/countries.ts (hunk 10 above)
Comment (jmrossy) [reply packages/utils/src/countries.ts]: utils/collections

File: packages/mobile/src/send/SendAmount.tsx [hunk 11]
```diff
@@ -155,9 +160,23 @@ export class SendAmount extends React.Component<Props, State> {
...
   }
 
+  getDollarsAmount = () => {
+    const parsedInputAmount = parseInputAmount(this.state.amount)
+
+    if (LOCAL_CURRENCY_SYMBOL) {
+      const { localCurrencyExchangeRate } = this.props
+      return (
+        convertLocalAmountToDollars(parsedInputAmount, localCurrencyExchangeRate) ||
+        new BigNumber('')
```
Comment (jmrossy) [inline packages/mobile/src/send/SendAmount.tsx:174]: why '' instead of 0 here?

File: packages/mobile/src/send/SendAmount.tsx (hunk 7 above)
Comment (jmrossy) [reply packages/mobile/src/send/SendAmount.tsx]: I'd prefer we just pass around all the information we need in it's original BigNumber form instead of mixing in this complicated decimal handling.

You're closer certainly to this feature than me so I'lll leave it to you to decide. If you do want to leave this rounding logic in place, please make a utility for it.

File: packages/mobile/src/send/SendAmount.tsx (hunk 11 above)
Comment (jeanregisser) [reply packages/mobile/src/send/SendAmount.tsx:174]: I wanted the equivalent to `NaN` if the exchange rate hasn't been fetched yet. 0 would be misleading.

File: packages/mobile/src/send/SendAmount.tsx (hunk 7 above)
Comment (jeanregisser) [reply packages/mobile/src/send/SendAmount.tsx]: I think this is the best place for now to handle this. As this is the screen where the local currency amount is entered and is the one dealing with the conversion.

This input is also used for the request confirmation screen, and passes the right value to handle the case I described.

I'll wrap it in a function and add the full comment above.

**Review by jmrossy — APPROVED** [review]
Looks good!
